src/ray/util/util.cc

```diff
@@ -362,3 +362,18 @@ std::shared_ptr<std::unordered_map<std::string, std::string>> ParseURL(std::stri
   result->emplace(key_value_pair.first, key_value_pair.second);
   return result;
 }
+
+void RunFnPeriodically(std::function<void()> fn, boost::posix_time::milliseconds period,
```

Comment:
Maybe we can change this to a helper class (say `PeriodicRunner`), so the callers can just pass in a function and a period, without having to be aware of the timers.

Reply:
great! I'll change that.